### PR TITLE
Implement Get mentors by category endpoint (Admin) #26

### LINKED ISSUE #26: Implement Get mentors by category endpoint (Admin)
**Description:**
This issue involves implementing an API to get mentors by category. The endpoint should allow Admin to make a GET request to `{{baseUrl}}/api/admin/mentors?category={{category}}`.

```
[
    {
        "mentorId": 2,
        "category": "Computer Science",
        "profile": {
            "contact_email": "thomas@gmail.com",
            "first_name": "Thomas",
            "last_name": "Daniel",
            "image_url": "https://www.linkedin.com/thomas",
            "linkedin_url": "https://www.linkedin.com/thomas"
        },
        "appilcation": {
            "position": "Software Engineer",
            "bio": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
            "additonalLinks": [
                "https://www.linkedin.com",
                "https://www.google_scholar.com"
            ],
            "slots": 10
        }
    },
    {
        "mentorId": 5,
        "category": "Computer Science",
        "profile": {
            "contact_email": "mandy@gmail.com",
            "first_name": "Mandy",
            "last_name": "Fernando",
            "image_url": "https://www.linkedin.com/mandy",
            "linkedin_url": "https://www.linkedin.com/mandy"
        },
        "appilcation": {
            "position": "Senior Software Engineer",
            "bio": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
            "additonalLinks": [
                "https://www.linkedin.com",
                "https://www.google_scholar.com"
            ],
            "slots": 10
        }
    }
]
```

**Tasks:**
1. Create a controller for the `/controllers/mentor` endpoint in the backend (create the route mentor if not created).
2. Parse and validate the request body to ensure it matches the expected format.
3. Implement appropriate error handling and response status codes for different scenarios (e.g., validation errors, database errors).
5. Write unit tests to validate the functionality and correctness of the endpoint.

API documentation: https://documenter.getpostman.com/view/27421496/2s93m1a4ac#8744a3ee-970f-489a-853d-8b23fdee8de3

ER diagram: https://drive.google.com/file/d/11KMgdNu2mSAm0Ner8UsSPQpZJS8QNqYc/view 

**Acceptance Criteria:**
- The profile API endpoint `/controllers/mentor` is implemented via a GET request.
- The request body is properly parsed and validated for the required format.
- Appropriate error handling is implemented, providing meaningful error messages and correct response status codes.
- Unit tests are written to validate the functionality and correctness of the endpoint.

**Additional Information:**
No

**Related Dependencies or References:**
No

### PR DESCRIPTION
**Pull request by sathudeva7**
## Purpose

The purpose of this PR is to fix #26

## Goals

This issue involves implementing an API to get mentors by category. The endpoint should allow Admin to make a GET request to {{baseUrl}}/api/admin/mentors?category={{category}}.

## Approach

`    const mentors: Mentor[] = await mentorRepository
      .createQueryBuilder('mentor')
      .innerJoinAndSelect('mentor.category', 'category')
      .leftJoinAndSelect('mentor.profile', 'profile')
      .select([
        'mentor.uuid',
        'category.category',
        'mentor.application',
        'profile.contact_email',
        'profile.first_name',
        'profile.last_name',
        'profile.image_url',
        'profile.linkedin_url'
      ])
      .where('category.category ILIKE :name', {
        name: categoryQuery
      })
      .getMany()`

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by sathudeva7** [comment]
@anjula-sack please review this

File: src/services/admin/mentor.service.ts [hunk 1]
```diff
@@ -109,3 +109,52 @@ export const findAllMentorEmails = async (
...
+      .select([
+        'mentor.uuid',
+        'category.category',
+        'mentor.application',
+        'profile.contact_email',
+        'profile.first_name',
+        'profile.last_name',
+        'profile.image_url',
+        'profile.linkedin_url'
+      ])
```
Comment (anjula-sack) [inline src/services/admin/mentor.service.ts:138]: Without selecting fields, what does it return? @sathudeva7  Also can you drop a comment on the issue and get assigned?

Comment (sathudeva7) [reply src/services/admin/mentor.service.ts:138]: sorry for the mistake could you please assign me to this issue

Comment (sathudeva7) [reply src/services/admin/mentor.service.ts:138]: without selecting fields ->

Screenshot 2023-10-24 at 16 27 54

Comment (anjula-sack) [reply src/services/admin/mentor.service.ts:138]: @sathudeva7 Can you drop a comment on the issue then I will assign you

Comment (anjula-sack) [reply src/services/admin/mentor.service.ts:138]: @sathudeva7 Can you update it to without selecting?

Comment (sathudeva7) [reply src/services/admin/mentor.service.ts:138]: i will update within this week

Comment (anjula-sack) [reply src/services/admin/mentor.service.ts:138]: Any updates @sathudeva7?

**Pull request closed without merging**